http/transport.go

```diff
@@ -2,13 +2,28 @@ package http
 
 import (
 	"crypto/tls"
+	"fmt"
 	"net"
 	"net/http"
+	"net/url"
+	"strings"
+	"syscall"
 	"time"
+
+	furl "github.com/influxdata/flux/dependencies/url"
 )
 
 // NewDefaultTransport creates a new transport with sane defaults.
-func NewDefaultTransport() *http.Transport {
+func NewDefaultTransport(dialer *net.Dialer) *http.Transport {
+
+	if dialer == nil {
+		dialer = &net.Dialer{
```

Comment:
Does this `dialer` get used anywhere else in this function? I was thinking it could be used down on line 30 instead of making a new one.